### DISCUSSION TITLE
Getting creation date of psn account

### DISCUSSION THREAD

**Discussion by WhiteShadow8500** [isFakeAccount/psnawp#48] [Ideas]
How to get creation date of psn account

**Comment by isFakeAccount** [comment]
I don't know the end point. If you can get this information from playstation app then I might be able to add it.

**Reply by Delta0S** [reply]
![image](https://github.com/isFakeAccount/psnawp/assets/156870537/36fc3cf1-166a-48e4-9a90-d80d766bb140) Not sure if this helps

**Reply by isFakeAccount** [reply]
> Not sure if this helps

It doesn't unfortunately. I suggest reading CONTRIBUTING.md in repo and see if you can find the endpoint that returns the account creation date. 

@Delta0S

**Reply by Delta0S** [reply]
@isFakeAccount 

How about this? would this help you?

![image](https://github.com/isFakeAccount/psnawp/assets/156870537/6311ef40-81c4-4657-9975-582bff2972ec)
![image](https://github.com/isFakeAccount/psnawp/assets/156870537/1f279dc0-4058-4c3d-904a-6297ad118151)
![image](https://github.com/isFakeAccount/psnawp/assets/156870537/7ef17cd0-002d-4b35-ac87-572c15efc797)
Does this help? The person behind initially used the [PSN API](https://psn-api.achievements.app/api-docs/users) and then customized it to work with just three files. It seems they use an iPhone for this. I'll also try following the CONTRIBUTING.md tutorials to see what happens.

**Reply by isFakeAccount** [reply]
@Delta0S I asked the psn-api dev. Lets see what they have. In documentation I don't see any reference to account creation date. 

https://github.com/achievements-app/psn-api/discussions/167

**Reply by IjiPaz** [reply]
found this a while ago and saved it. from playstation direct, not from ps app. no idea about other ways to get it and do not have more info

```
headers = {
    'accept': 'application/json, text/plain, */*',
    'accept-language': 'en-GB,en-US;q=0.9,en;q=0.8,de-DE;q=0.7,de;q=0.6',
    'authorization': f'Bearer {bearer_token}', 
    'cache-control': 'no-cache',
    'content-type': 'application/json',
    'dnt': '1',
    'origin': 'https://direct.playstation.com',
    'pragma': 'no-cache',
    'priority': 'u=1, i',
    'referer': 'https://direct.playstation.com/',
    'sec-ch-ua': '"Not)A;Brand";v="8", "Chromium";v="138"',
    'sec-ch-ua-mobile': '?0',
    'sec-ch-ua-platform': '"Windows"',
    'sec-fetch-dest': 'empty',
    'sec-fetch-mode': 'cors',
    'sec-fetch-site': 'same-site',
    'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36',
}

params = {
    'fields': 'FULL',
}

json_data = {
    'userId': f'{hashed_accountid}',
}

response = requests.post(
    'https://api.direct.playstation.com/commercewebservices/ps-direct-de/my-account/profile',
    params=params,
    headers=headers,
    json=json_data,
)
```

response, customerSince is account creation date
```
{
    "uid": "",
    "firstName": "",
    "lastName": "",
    "displayUid": "",
    "avatarUrl": "",
    "email": "",
    "contactNumber": "",
    "customerSince": "yyyy-dd-mmThh:mm:ss.020Z",
    "psPlus": ,
    "country": "",
    "language": "",
    "banned": false
}
```
